java-checks/src/main/java/org/sonar/java/checks/ConstantMethodCheck.java

```diff
@@ -62,6 +65,15 @@ public void visitNode(Tree tree) {
     }
   }
 
+  private static boolean hasDefaultKeyword(MethodTreeImpl methodTree) {
+    Symbol.MethodSymbol symbol = methodTree.symbol();
```

Comment:
There is a way simpler approach to check if it's a default method (see here: [UnusedMethodParameterCheck.java#L119](https://github.com/SonarSource/sonar-java/blob/4.4.0.8066/java-checks/src/main/java/org/sonar/java/checks/unused/UnusedMethodParameterCheck.java#L119)):
```
ModifiersUtils.hasModifier(modifiers, Modifier.DEFAULT)
```